src/main/kotlin/org/rust/ide/inspections/fixes/RemoveRefFix.kt

```diff
@@ -31,7 +36,26 @@ class RemoveRefFix(
         if (argEl.text != null && argEl.text[0] == '&') {
             val offset = argEl.startOffset
             val document = PsiDocumentManager.getInstance(project).getDocument(argEl.containingFile)
-            document?.deleteString(offset, offset + 1)
+
+            // 1. If the reference is a mut one (&mut X) we want to remove the `mut` part too as `mut X` is either not
```

Comment:
This code is too complex because the author of the original quickfix decided to work with a text instead of PSI. If work with PSI, the implementation of `invoke` may looks like this
```rust
if (argEl is RsUnaryExpr && argEl.operatorType in listOf(UnaryOperator.REF, UnaryOperator.REF_MUT)) {
    argEl.replace(argEl.expr)
}
// yes, this is entire impl of `invoke` =/
```

Reply:
Thanks! Totally forgot I could just do that